backend/app/services/git_skill_service.py

```
@@ -0,0 +1,1046 @@
+    try:
+        metadata = yaml.safe_load(yaml_content)
+        if not isinstance(metadata, dict):
+            raise ValueError("YAML frontmatter must be a dictionary")
+
+        if "description" not in metadata:
+            raise ValueError("SKILL.md frontmatter must include 'description' field")
+
+        return metadata
+
+    except yaml.YAMLError as e:
+        raise ValueError(f"Invalid YAML frontmatter: {str(e)}")
```

Ruff (0.14.14)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)